keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -244,21 +261,41 @@ We expect no non-infra related flakes in the last month as a GA graduation crite
   Once CRI implementation provide the relevant RPC calls
   the e2e tests will not fail but need to be extended.
 
+- Once the initial Alpha release  CRI-O supports the
+  `CheckpointContainer` CRI RPC and tests have been
+  enhanced to support CRI implementation that implement
+  the `CheckpointContainer` CRI RPC
+
+- Once Kubernetes was released with the `CheckpointContainer` CRI RPC
+  CRI-O has been updated to support the new CRI RPC.
+  The tests have been enhanced to work with CRI implementations
+  that support the `CheckpointContainer` CRI RPC as well as
+  CRI implementations that do not support it. The tests also handle
+  if the corresponding feature gate is disabled or enabled:
+  <https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/checkpoint_container.go>
+
 ### Graduation Criteria
 
 #### Alpha
 
-- [ ] Implement the new feature gate and kubelet implementation
-- [ ] Ensure proper tests are in place
-- [ ] Update documentation to make the feature visible
+- [X] Implement the new feature gate and kubelet implementation
+- [X] Ensure proper tests are in place
+- [X] Update documentation to make the feature visible
+  - <https://kubernetes.io/docs/reference/node/kubelet-checkpoint-api/>
+  - <https://kubernetes.io/blog/2022/12/05/forensic-container-checkpointing-alpha/>
+  - <https://kubernetes.io/blog/2023/03/10/forensic-container-analysis/>
 
 #### Alpha to Beta Graduation
```

Comment:
This ought to be testable by running a privileged container and checkpointing and restoring a particular pod, right?

Reply:
Complicated question. First, I am not sure why you mention "privileged container" here. GitHub tells me you are commenting on `#### Alpha to Beta Graduation`, which does not mention "privileged container". So that confuses me, sorry.

Then you talk about "checkpointing and restoring a particular pod". Although restoring is possible it is done by the container engine and Kubernetes is not involved in restoring so far. Then you mention "pod", currently it is only about containers. Also confused.

Don't know what to answer here.

Reply:
In cluster restore is out of scope for this one, but yes, a privileged pod could be used to trigger a checkpoint and check if it got created successfully.

Reply:
Sorry, I still have problems understand the need for a privileged container. There are already unit tests and e2e tests available since the initial introduction of the feature. What is the question here?

Reply:
Glad to hear there are tests.  This section should indicate exactly which e2e tests exist. Usually by pointing to test-grid to show they also run.  See https://github.com/kubernetes/enhancements/pull/4288/files#r1480513382

Reply:
I am having trouble finding the tests. The tests are only run in the CI run that enabled all Alpha Feature Gates.  Which does not run always if I understand the setup correctly. This is really complicated to find.

Can anyone point me where the Alpha Tests are being recorded in the test grid.

Reply:
There isn't really a good job to look at for alpha tests. I think your tests may be getting skipped right now.

Usually people add alpha jobs and test those separately.

Reply:
I see the test getting called in this job: https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-features

But it is being skipped.

Reply:
Yes, it is being skipped. But during initial merge of the feature someone triggered a test which was called something like "All-Alpha-Feature-Enabled". I was also trying to find to tests, but I never found them later.

Reply:
https://testgrid.k8s.io/sig-node-presubmits is probably what someone used.

I don't see a crio alpha job after all. I see a containerd one but that job is failing due to the lack of containerd implementation.